frontend/javascripts/oxalis/model/bucket_data_handling/data_cube.js

```diff
@@ -259,17 +260,22 @@ class DataCube {
 
   addBucketToGarbageCollection(bucket: DataBucket): void {
     if (this.bucketCount >= this.MAXIMUM_BUCKET_COUNT) {
-      for (let i = 0; i < 2 * this.bucketCount; i++) {
```

Comment:
I think, earlier versions of `shouldCollect()` had side effects which is why this code iterated twice over all buckets. However, now, the method is sideeffect-free which is why a single iteration should be enough.